src/server/replica.cc

```diff
@@ -851,13 +851,47 @@ void Replica::RedisStreamAcksFb() {
   }
 }
 
+class AclCheckerClient : public ProtocolClient {
+ public:
+  AclCheckerClient(ServerContext server, Context* cntx)
+      : ProtocolClient(std::move(server)), cntx_(cntx) {
+    Connect();
+  }
+
+  void CheckAclRoundTrip() {
+    if (auto ec = SendCommandAndReadResponse(StrCat("REPLCONF acl-check ", "0")); ec) {
+      cntx_->Cancel();
+      LOG(INFO) << "Error in REPLCONF acl-check: " << ec.message();
+    } else if (!CheckRespIsSimpleReply("OK")) {
+      cntx_->Cancel();
+      LOG(INFO) << "Error: " << ToSV(LastResponseArgs().front().GetBuf());
+    }
+  }
+
+ private:
+  void Connect() {
+    VLOG(1) << "Connecting with acl client";
+    auto ec = ConnectAndAuth(absl::GetFlag(FLAGS_master_connect_timeout_ms) * 1ms, cntx_);
```

Comment:
When you create this connection in rare cases it could get another user and password from the replication connections right? because we can set them on run time.
So it could be that the replications acks are rejected and this checker is passing

Reply:
> When you create this connection in rare cases it could get another user and password from the replication connections

Well you need to think about the use care here: 

1. We call config set for masterauth 
2. We call config set for masteruser
3. We call REPLICAOF ....

There is a single actor here and a chore of 3 steps (1 to 3 above). If there were more, you probably doing something really really wrong. Would you ever spin up a replica and allow different people to start replication under a different user ? In production environments I would not expect that ever. And even if this is the case, a) it will fail based on timing b) if you change the masteruser/masterauth `after` you initiate replication you `should` restart it. 

I could in theory cache the user/password and reconnect if it ever changes but this is sneaky and the semantics are not very clear so I don't allow this.